panic: runtime error: invalid memory address or nil pointer dereference

### Background

The `panic: runtime error: invalid memory address or nil pointer dereference` error. It's `lnd-38`

### Your environment

* version of `lnd` - 0.9.0-beta commit = **v0.9.0-beta-26-g51d8c11eb5ade74486786429de3913655ceba7f1**
* which operating system (`uname -a` on *Nix) - CentOS 6.*

### Steps to reproduce

I don't know

### Expected behaviour

To work without crash

### Actual behaviour

```
2020-02-07 09:57:02.790 [INF] DISC: Creating new GossipSyncer for peer=03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658
2020-02-07 09:57:02.947 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): filtering chan range: start_height=0, num_blocks=2147483647
2020-02-07 09:57:02.955 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): sending range chunk of size=8000
2020-02-07 09:57:03.048 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): applying new update horizon: start=2020-02-07 09:57:02 +0100 CET, end=2088-02-25 13:11:09 +0100 CET, backlog_size=0
2020-02-07 09:57:03.048 [INF] HSWC: ChannelLink(616345:793:0): received re-establishment message from remote side
2020-02-07 09:57:03.137 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): sending range chunk of size=8000
2020-02-07 09:57:03.227 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): sending range chunk of size=8000
2020-02-07 09:57:03.317 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): sending range chunk of size=8000
2020-02-07 09:57:03.409 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): sending final chan range chunk, size=81
2020-02-07 09:57:03.860 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): fetching chan anns for 100 chans
2020-02-07 09:57:04.415 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): fetching chan anns for 100 chans
2020-02-07 09:57:05.003 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): fetching chan anns for 100 chans
2020-02-07 09:57:05.560 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): fetching chan anns for 100 chans
2020-02-07 09:57:06.128 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): fetching chan anns for 100 chans
2020-02-07 09:57:06.727 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): fetching chan anns for 100 chans
2020-02-07 09:57:07.379 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): fetching chan anns for 100 chans
2020-02-07 09:57:07.978 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): fetching chan anns for 100 chans
2020-02-07 09:57:08.535 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): fetching chan anns for 100 chans
2020-02-07 09:57:09.196 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): fetching chan anns for 100 chans
2020-02-07 09:57:09.771 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): rate limiting gossip replies, responding in 3.175808546s
2020-02-07 09:57:12.947 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): fetching chan anns for 100 chans
2020-02-07 09:57:13.733 [INF] DISC: GossipSyncer(03e3bfc9b4cbd8ce862278ac76234ed1aad6d83fdd29b5305c55c361452df42658): rate limiting gossip replies, responding in 4.213762401s
2020-02-07 09:57:14.632 [INF] HSWC: ChannelLink(616345:793:0): exited
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xad2f8b]

goroutine 5507077 [running]:
container/list.(*List).Remove(...)
        /home/bitcoin/go1.13.3/src/container/list/list.go:140
github.com/lightningnetwork/lnd/lnwallet.(*updateLog).removeUpdate(...)
        /home/bitcoin/go/src/github.com/lightningnetwork/lnd/lnwallet/channel.go:1094
github.com/lightningnetwork/lnd/lnwallet.compactLogs.func1(0xc00356c8a0, 0xc00356c750)
        /home/bitcoin/go/src/github.com/lightningnetwork/lnd/lnwallet/channel.go:1162 +0xfb
github.com/lightningnetwork/lnd/lnwallet.compactLogs(0xc00356c750, 0xc00356c8a0, 0x3, 0x1)
        /home/bitcoin/go/src/github.com/lightningnetwork/lnd/lnwallet/channel.go:1177 +0x90
github.com/lightningnetwork/lnd/lnwallet.(*LightningChannel).ReceiveRevocation(0xc00dfe08f0, 0xc004755cc0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/bitcoin/go/src/github.com/lightningnetwork/lnd/lnwallet/channel.go:4509 +0x1306
github.com/lightningnetwork/lnd/htlcswitch.(*channelLink).handleUpstreamMsg(0xc005763d40, 0x15eef60, 0xc004755cc0)
        /home/bitcoin/go/src/github.com/lightningnetwork/lnd/htlcswitch/link.go:1787 +0xd79
github.com/lightningnetwork/lnd/htlcswitch.(*channelLink).htlcManager(0xc005763d40)
        /home/bitcoin/go/src/github.com/lightningnetwork/lnd/htlcswitch/link.go:1127 +0x847
created by github.com/lightningnetwork/lnd/htlcswitch.(*channelLink).Start
        /home/bitcoin/go/src/github.com/lightningnetwork/lnd/htlcswitch/link.go:475 +0x4a8
```

--- a/lnwallet/channel.go
+++ b/lnwallet/channel.go
@@ -1064,7 +1064,7 @@ func (u *updateLog) appendUpdate(pd *PaymentDescriptor) {
 // also added to index accordingly. This function differs from appendUpdate in
 // that it won't increment the log index counter.
 func (u *updateLog) restoreUpdate(pd *PaymentDescriptor) {
-	u.updateIndex[u.logIndex] = u.PushBack(pd)
+	u.updateIndex[pd.LogIndex] = u.PushBack(pd)
 }
 
 // appendHtlc appends a new HTLC offer to the tip of the update log. The entry

--- a/lnwallet/channel_test.go
+++ b/lnwallet/channel_test.go
@@ -3146,7 +3146,7 @@ func TestChanSyncOweCommitmentPendingRemote(t *testing.T) {
 	}
 
 	// Bob signs the commitment he owes.
-	_, bobHtlcSigs, _, err := bobChannel.SignNextCommitment()
+	bobCommit, bobHtlcSigs, _, err := bobChannel.SignNextCommitment()
 	if err != nil {
 		t.Fatalf("unable to sign commitment: %v", err)
 	}
@@ -3154,6 +3154,20 @@ func TestChanSyncOweCommitmentPendingRemote(t *testing.T) {
 	// This commitment is expected to contain no htlcs anymore.
 	if len(bobHtlcSigs) != 0 {
 		t.Fatalf("no htlcs expected, but got %v", len(bobHtlcSigs))
+	}
+
+	// Get Alice to revoke and trigger Bob to compact his logs.
+	err = aliceChannel.ReceiveNewCommitment(bobCommit, bobHtlcSigs)
+	if err != nil {
+		t.Fatal(err)
+	}
+	aliceRevoke, _, err := aliceChannel.RevokeCurrentCommitment()
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, _, _, _, err = bobChannel.ReceiveRevocation(aliceRevoke)
+	if err != nil {
+		t.Fatal(err)
 	}
 }
 

